galley/pkg/config/analysis/analyzers/injection/injection.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package injection
+
+import (
+	"strings"
+
+	"istio.io/istio/galley/pkg/config/analysis"
+	"istio.io/istio/galley/pkg/config/analysis/msg"
+	"istio.io/istio/galley/pkg/config/processor/metadata"
+	"istio.io/istio/galley/pkg/config/resource"
+	v1 "k8s.io/api/core/v1"
+)
+
+// Analyzer checks conditions related to Istio injection
+type Analyzer struct{}
+
+var _ analysis.Analyzer = &Analyzer{}
+
+// Name implements Analyzer
+func (s *Analyzer) Name() string {
+	return "injection.Analyzer"
+}
+
+// We assume that enablement is via an istio-injection=enabled namespace label
+// In theory, there can be alternatives using Mutatingwebhookconfiguration, but they're very uncommon
+// See https://istio.io/docs/ops/troubleshooting/injection/ for more info.
+const injectionLabelName = "istio-injection"
+const injectionLabelEnableValue = "enabled"
+
+// Analyze implements Analyzer
+func (s *Analyzer) Analyze(c analysis.Context) {
+	injectedNamespaces := make(map[string]bool)
+
+	c.ForEach(metadata.K8SCoreV1Namespaces, func(r *resource.Entry) bool {
+
+		// Ignore system namespaces
+		if strings.HasPrefix(r.Metadata.Name.String(), "kube-") || strings.HasPrefix(r.Metadata.Name.String(), "istio-") {
```

Comment:
This probably needs to be configurable.

Reply:
@sushicw I think we will need to add a configuration mechanism for the analyzers after all.

Reply:
I think there is a fine line of amount of flexibility we want to allow, vs making the 99% case work. And I wouldn't want to ask the user to input this. Instead, ideally by reflecting on the cluster we would figure out what the namespaces are. But it's probably non-trivial.

Reply:
I don't think we have a rule that says istio should be deployed to namespaces with istio-* prefix. When we allow deploying multiple control planes on the same cluster, this is likely to be a more prominent problem. So, a certain level of ambient context can certainly help the analyzers to be less brittle in general.

We don't need to fix this in this PR. For the time being, we can put a TODO here to at least track and solve the problem in the future. (An associated bug would also help for tracking).

Reply:
Ok, added TODO.

Reply:
We're probably fine to just skip `kube-system` namespace here. Other namespaces should be selected/skipped with the label/namespaceSelector method.

Reply:
Agreed. Created https://github.com/istio/istio/issues/17195. David, please reference the issue in your comment.